packages/react/src/components/popover/popover.tsx

```diff
@@ -26,24 +26,39 @@ export interface PopoverProps extends BaseProps, Omit<Div, 'ref'> {
   target?: RefObject<Element>;
 }
 
-export const Popover = ({
+export function Popover({
   align = 'center',
   onClose,
   position = 'top',
   target,
   ...props
-}: PopoverProps) =>
-  target ? (
+}: PopoverProps) {
+  const popover = useRef<HTMLDivElement>(null);
+  const [placement, setPlacement] = useState([position, align] as const);
+
+  useIntersectionObserver(
+    (entry) => setPlacement((placement) => optimalPlacement(entry, placement)),
+    [popover]
+  );
+
```

Comment:
[Minor] To increase readability a little, I would suggest:

```js
const [position, align] = placement;
```

Reply:
I see, right.

What if these then are `optimalPosition` and `optimalAlign`?

Minor though, up to you.

Reply:
I like that. Applied.